packages-rs/drainpipe/src/main.rs

```diff
@@ -143,16 +157,16 @@ async fn main() {
                         }
                         Err(error) => {
                             eprintln!("Error processing message: {error:?}");
-                            if let ProcessError::ProcessError { seq, error } = error {
-                                record_dead_letter(
-                                    &mut ctx.db_connection,
-                                    // TODO: Would be good to include the actual dropped message here but my rust skill issues are preventing me from doing so
-                                    &error.to_string(),
-                                    seq,
-                                )
-                                .map_err(|_| eprintln!("Failed to record dead letter"))
-                                .ok();
-                            }
+                            record_dead_letter(
+                                &mut ctx.db_connection,
+                                error.kind,
+                                &error.inner.to_string(),
+                                error.seq,
+                                &String::from_utf8(error.msg)
+                                    .unwrap_or("Unable to decode message".to_string()),
+                            )
```

Comment:
This will transform the `Vec<u8>` to a `String` for the DB, and replace it with this placeholder string if `from_utf8` throws an error.  might want to do something else instead though? convert the bytes to a hex string or something?